drivers/i2c/Kconfig

```diff
@@ -90,19 +90,26 @@ config I2C_NRF5_GPIO_SCL_PIN
 	help
 	  The GPIO pin to use for SCL.
 
-config I2C_STM32LX
-	bool "STM32Lx MCU I2C Driver"
+config I2C_STM32_V1
+	bool "STM32 V1 Driver (F1/L1/F4X)"
+	depends on SOC_FAMILY_STM32 && SOC_SERIES_STM32F4X
```

Comment:
wouldn't it be better if we change the depends when we add the i2c support (pinmux driver changes, soc pinmux changes etc) on the rest of the families?